src/main/scala/is/hail/utils/richUtils/RichHadoopConfiguration.scala

```diff
@@ -29,7 +29,7 @@ class RichHadoopConfiguration(val hConf: hadoop.conf.Configuration) extends AnyV
       os
   }
 
-  private def open(filename: String): InputStream = {
+  def open(filename: String): InputStream = {
```

Comment:
> Doesn't it seem like classic non-orthoganlity to have a separate method for each type of resource

Yes, but I preferred having an interface where it is impossible to do the wrong thing.  Some sloppy code snuck in early on and by making open and create private we caught them all. 
 We can fix this by having a `RequiresClosing[T]` which supports stacking of stream types and can only be consumed by `using`, but unused values will still be unclosed.  I guess that isn't so bad.  Someone could also try to sneak in a call to the IO primitives themselves.